metadata-ingestion/src/datahub/ingestion/source/schema_inference/json.py

```diff
@@ -33,14 +33,31 @@
 
 
 class JsonInferrer(SchemaInferenceBase):
+    def __init__(self, max_rows: int = None, format: str = "json"):
+        self.max_rows = max_rows
+        self.format = format
+
     def infer_schema(self, file: IO[bytes]) -> List[SchemaField]:
-        try:
-            datastore = ujson.load(file)
-        except ujson.JSONDecodeError as e:
-            logger.info(f"Got ValueError: {e}. Retry with jsonlines")
+        if self.format == "jsonl":
             file.seek(0)
             reader = jsl.Reader(file)
-            datastore = [obj for obj in reader.iter(type=dict, skip_invalid=True)]
+            datastore = []
```

Comment:
can make this simpler using itertools

```suggestion
            datastore = [obj for obj in itertools.islice(reader.iter(type=dict, skip_invalid=True), 0, 100)]
```

Reply:
Thanks for the suggestion. I'll update the same. I must confess, I'm more of a Python enthusiast than a Python purist!